permissions/src/main/java/com/google/accompanist/permissions/PermissionState.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.permissions
+
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.ActivityResultRegistry
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+
+/**
+ * Creates a [PermissionState] that is remembered across compositions.
+ *
+ * It's recommended that apps exercise the permissions workflow as described in the
+ * [documentation](https://developer.android.com/training/permissions/requesting#workflow_for_requesting_permissions).
+ *
+ * @param permission the permission to control and observe.
+ */
+@Composable
+fun rememberPermissionState(
+    activityResultRegistry: ActivityResultRegistry,
```

Comment:
I'd suggest a version that uses a default value pulled from [LocalActivityResultRegistryOwner](https://developer.android.com/reference/kotlin/androidx/activity/compose/LocalActivityResultRegistryOwner) as the source to get the correct `ActivityResultRegistry`.

Reply:
Added! Thank you :) much better know

Reply:
Do we need the explicit overload? Could this not just be a default param value?

`activityResultRegistry: ActivityResultRegistry = LocalActivityResultRegistryOwner.current`

We'd need to flip the param order, but this is better imo.